src/game_object.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include "scene/scene_node.h"
+#include "model/model.h"
+
+/*
+	This represents an abstract object in the scene.
+	Contains all the essential components required to
+	render the model in the scene.
+*/
+class GameObject :
```

Comment:
Put inheritance on same line.

Also, at the moment, this seems to be a thin wrapper over `Model`. I assume you're going to add more things to this, though.